tests/ssh_tests.py

```diff
@@ -83,13 +85,23 @@ def trying_to_use_ssh_shell_after_exit_results_in_error():
     assert_raises(Exception, lambda: shell.run(["true"]))
 
 
-def _create_shell_with_wrong_port():
+@istest
+def passing_a_custom_socket_like_a_pre_connected_one():
+    sock = socket.socket(socket.AddressFamily.AF_INET, socket.SOCK_STREAM)
+    retry_on_signal(lambda: sock.connect((HOSTNAME, PORT)))
+
+    with _create_shell_with_wrong_port(sock=sock) as shell:
+        shell.run(["true"])
+
+
+def _create_shell_with_wrong_port(sock=None):
     return spur.SshShell(
-        username="bob",
-        password="password1",
-        hostname="localhost",
+        username=USERNAME,
+        password=PASSWORD,
+        hostname=HOSTNAME,
         port=54321,
         missing_host_key=spur.ssh.MissingHostKey.accept,
+        sock=sock
```

Comment:
To clarify: I'm not suggesting that the kwargs get passed to paramiko, just for this test function, so you'd end up with:

```python
def _create_shell_with_wrong_port(**kwargs):
    return spur.SshShell(
        username=USERNAME,
        password=PASSWORD,
        hostname=HOSTNAME,
        port=54321,
        missing_host_key=spur.ssh.MissingHostKey.accept,
        **kwargs
    )
```

That way, the other usages of `_create_shell_with_wrong_port` continue to pass no value for `sock`, rather than passing a value of `None`, which is closer to common usage. I think having an explicit `sock` argument in the actual constructor for `SshShell` is entirely correct.